rust/src/http2/parser.rs

```diff
@@ -763,7 +763,13 @@ pub fn http2_parse_frame_settings(i: &[u8]) -> IResult<&[u8], Vec<HTTP2FrameSett
 
 pub fn doh_extract_request(i: &[u8]) -> IResult<&[u8], Vec<u8>> {
     let (i, _) = tag("/dns-query?dns=")(i)?;
-    match general_purpose::STANDARD.decode(i) {
+    let mut needed_padding = 4 - (i.len() % 4);
+    let mut padded_vec = i.to_vec();
+    while needed_padding > 0 {
+        padded_vec.push(b'=');
```

Comment:
This code should be in some util file, not specific for http2, which should just call some base64 decoder

Reply:
Why not use `STANDARD_NO_PAD` ? cf https://docs.rs/base64/latest/base64/engine/general_purpose/constant.STANDARD_NO_PAD.html

Reply:
Nice. Thank you. Doing in the next rev